### DISCUSSION TITLE
Why you use goto?

### DISCUSSION THREAD

**Discussion by 64J0** [duckdb/duckdb#13970] [Q&A] [answered]
Is there any performance advantage for using goto instead of calling functions? For example: [link](https://github.com/duckdb/duckdb/blob/main/src/parser/parser.cpp#L152).

I know that there is that old paper from Dijkstra that states that goto is considered harmful. So I'd like to know why you decided to use it when building this tool.

**Comment by arjenpdevries** [comment, marked as answer]
This is a nice overview:
https://langdev.stackexchange.com/questions/581/what-are-the-pros-and-cons-of-goto
